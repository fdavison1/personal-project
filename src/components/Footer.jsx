import React from 'react'
import styled from 'styled-components'
import linked from '../assets/In.png'
import github from '../assets/25231.png'

const Stripe = styled.div`
display: flex
position: relative
bottom: 0
flex-direction: column
align-items: center
background: #c2dde6
box-sizing: border-box
position: absolute
width: 100%
height: 100px`


const Foot = styled.div`
margin: 10px
padding: 0 50px
height: 100px
box-sizing: border-box
display: flex
align-items: center
justify-content: space-between
width: 1100px`

const FootLogo = styled.img`
height: 50px
margin: 0 10px`

const LogoBox = styled.div`
display: flex
align-items: center`

const Paragraph = styled.p`
font-size: 1.2rem
font-weight: 200
`

export default function Footer(){
    return (
        <Stripe>

        <Foot>

        <Paragraph>
        <span>Fred Davison</span> DevMountain Student 2019
        </Paragraph>

        <LogoBox>
            <FootLogo src={linked} alt="linkedIn logo"/>
            <FootLogo src={github} alt="github logo"/>
        </LogoBox>


        </Foot>

    </Stripe>
    )
}