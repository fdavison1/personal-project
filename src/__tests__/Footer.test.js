import React from 'react'
import { shallow } from '../enzyme'
import 'jest-styled-components'
import renderer from 'react-test-renderer'
import Footer, { Stripe } from '../components/Footer'


describe ('Footer.jsx', () => {
    it('renders', ()=> {
    
    const wrapper = shallow(<Footer/>)

    expect(wrapper.exists()).toBe(true)
    })
    it('matches snapshot', () => {
        const tree = renderer.create(<Footer />).toJSON()
        expect(tree).toMatchSnapshot()
    })
    it('Footer css', () => {
        const tree = renderer.create(<Stripe />).toJSON()
        expect(tree).toMatchSnapshot()
        expect(tree).toHaveStyleRule('background', '#c2dde6')
    })
})