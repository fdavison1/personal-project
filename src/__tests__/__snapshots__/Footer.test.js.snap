// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer.jsx Footer css 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: relative;
  bottom: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: #c2dde6;
  box-sizing: border-box;
  position: absolute;
  width: 100%;
  height: 100px;
}

<div
  className="c0"
/>
`;

exports[`Footer.jsx matches snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: relative;
  bottom: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: #c2dde6;
  box-sizing: border-box;
  position: absolute;
  width: 100%;
  height: 100px;
}

.c1 {
  margin: 10px;
  padding: 0 50px;
  height: 100px;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  width: 1100px;
}

.c4 {
  height: 50px;
  margin: 0 10px;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2 {
  font-size: 1.2rem;
  font-weight: 200;
}

@media (max-width:1025px) {
  .c1 {
    width: 800px;
  }
}

@media (max-width:500px) {
  .c1 {
    width: 100vw;
    -webkit-box-pack: space-evenly;
    -webkit-justify-content: space-evenly;
    -ms-flex-pack: space-evenly;
    justify-content: space-evenly;
    margin: 0;
  }
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <p
      className="c2"
    >
      <span>
        Fred Davison
      </span>
       DevMountain Student 2019
    </p>
    <div
      className="c3"
    >
      <a
        href="https://www.linkedin.com/in/fred-davison-220425196/"
        rel="noopener noreferrer"
        target="_blank"
      >
        <img
          alt="linkedIn logo"
          className="c4"
          src="In.png"
        />
      </a>
      <a
        href="https://github.com/fdavison1"
        rel="noopener noreferrer"
        target="_blank"
      >
        <img
          alt="github logo"
          className="c4"
          src="25231.png"
        />
      </a>
    </div>
  </div>
</div>
`;
